opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -136,22 +135,25 @@ impl LoggerProvider {
     }
 }
 
+impl Drop for LoggerProvider {
+    fn drop(&mut self) {
```

Comment:
I believe we want to replicate the Metrics behavior here as well https://github.com/open-telemetry/opentelemetry-rust/pull/2156/files 
But this looks different. Can you explain why is this differently handled? Specifically, why is Drop implemented for LoggerProvider as opposed to doing it to the inner one, given LoggerProvider is cloned for each logger?

Also, this still causes the false alarm to the user, when shutdown is triggered by Drop even when user has shutdown it already?

Reply:
yes good catch. I tried to optimize it incorrectly. it. Have added the validation in LoggerProviderInner::drop now.

Reply:
Also updated the unit-test to test with multiple LoggerProvider instances , sharing a clone of LoggerProviderInner.